src/annotator/util/buckets.js

```diff
@@ -0,0 +1,248 @@
+import { getBoundingClientRect } from '../highlighter';
+
+/**
+ * @typedef {import('../../types/annotator').Anchor} Anchor
+ */
+
+/**
+ * A structured Array representing either the top (`startOrEnd` = 1) or the
+ * bottom (`startOrEnd` = -1) of an anchor's highlight bounding box.
+ *
+ * @typedef {[pixelPosition: number, startOrEnd: (-1 | 1), anchor: Anchor]} PositionPoint
```

Comment:
The structure of a `PositionPoint` is not entirely intuitive but I'm leaving as-is for the moment.